integrations/segmentio/lib/index.js

```diff
@@ -315,10 +317,31 @@ Segment.prototype.normalize = function(message) {
   }
   if (this.options.addBundledMetadata) {
     var bundled = keys(this.analytics.Integrations);
+    var maybeBundledConfigIds = this.options.maybeBundledConfigIds
+
+    // Generate a list of bundled config IDs using the intersection of
+    // bundled destination names and maybe bundled config IDs.
+    var bundledConfigIds = []
+    for (var i = 0; i < bundled.length; i++) {
+      var name = bundled[i]
+      if (!maybeBundledConfigIds) {
```

Comment:
I think it would be clearer if this conditional was outside of the loop since it does not depend on anything in `bundled`.

Reply:
We'd have to refactor and wrap the for-loop in a conditional. I can go either way but was trying to avoid nested conditionals.

```
if (maybeBundledConfigIds) {
  for (var i = 0; i < bundled.length; i++) { ... }
}
```